### DISCUSSION TITLE
Duplicate tests for placing brackets at the end of a command line

### DISCUSSION THREAD

**Discussion by Drako2711** [pytest-dev/pytest#9772] [Q&A]
Hi, I am using the pytest command:
`python3 -m pytest LoginTest.py --alluredir ./results `
Console:
```
platform win32 -- Python 3.9.10, pytest-7.0.1, pluggy-1.0.0
rootdir: D:\PROYECTOS\SELENIUM\Pruebas
plugins: allure-pytest-2.9.45
collected 7 items
```
to go through the tests in my file, but when I add square brackets the items found are duplicated, for example from 7 it goes to 14, and so on.
This is the command, even if I leave the square brackets empty, it duplicates the same, has anyone else had this happen?
`python3 -m pytest PurchaseTest.py --alluredir ./results [var]`
Console:
```
platform win32 -- Python 3.9.10, pytest-7.0.1, pluggy-1.0.0
rootdir: D:\PROYECTOS\SELENIUM\Pruebas
plugins: allure-pytest-2.9.45
collected 14 items
```

**Comment by portyu9** [comment]
Hi @Drako2711 

`[var]` by itself is not pytest syntax for selecting a parametrized value.

A parametrized test case is selected using its complete **node ID**, for example:

```bash
pytest "PurchaseTest.py::test_purchase[var]"
```

Quoting the full node ID is especially important in shells where `[` and `]` have special meaning.

If `var` is intended to be an independent value passed to your test run, define a real command-line option instead, for example:

```python
def pytest_addoption(parser):
    parser.addoption("--var")
```

and invoke:

```bash
pytest PurchaseTest.py --alluredir ./results --var value
```

From the posted information I cannot prove exactly why the old invocation collected seven extra tests — shell parsing or a plugin handling that extra positional argument could be involved. `pytest --collect-only -vv ...` is the best way to compare exactly which node IDs each invocation is producing.

Please mark this answer as accepted if it helped, thank you.
